### PR TITLE
Applicant account - front end

### PR DESCRIPTION
**Pull request by SibingWu**
[Stretch]

Front end section
* Applicant account info page
image
* Edit account page
image

Still need double check, please do NOT review it for now:)

### CODE REVIEW

File: walk-in-interview/src/main/webapp/account/applicant/edit/index.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        <li>
+          <label for="skills-title"></label>
+          <div id="skills-title" name="skills-title"></div>
+          <ul id="skills"></ul>
+        </li>
+      </ul>
+    </div>
+
+    <div class="error-message" id="error-message"></div>
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/account/applicant/edit/index.html:29]: Is it intentional that this error locates between skill title and skill list?

File: walk-in-interview/src/main/webapp/account/applicant/edit/script.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+ */
+function getApplicantDetails() {
+  const queryString = window.location.search;
+
+  const urlParams = new URLSearchParams(queryString);
+  if (urlParams === '') {
+    throw new Error('url params should not be empty');
+  }
+
+  return JSON.parse(urlParams.get(APPLICANT_DETAILS_PARAM));
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/account/applicant/edit/script.js:82]: how can url param become detail?

File: walk-in-interview/src/main/webapp/account/applicant/edit/script.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+* Validates the user input.
+* Shows error message on the webpage if there is field with invalid input.
+*
+* @return {boolean} depending on whether the input is valid or not.
+*/
+function validateRequiredUserInput() {
+  const applicantId = getApplicantId();
+  const name = document.getElementById('name').value;
+
+  if (applicantId === '') {
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/account/applicant/edit/script.js:163]: Do user enter applicant id? Maybe this is something wrong rather than invalid user input?

File: walk-in-interview/src/main/webapp/account/applicant/script.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+window.onload = () => {
+  let accountId = getId();
+  renderPageElements(accountId);
+};
+
+/** Gets the ID of this account. */
+function getId() {
+  // TODO
+  return 'xxxxx';
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/account/applicant/script.js]: return 'not implemented'?

File: walk-in-interview/src/main/webapp/account/applicant/script.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+      'English',
+    ],
+  };
+
+  // if there is no details related to such id, return empty details
+  // (TODO)
+  return accountDetails;
+}
+
+function renderSkills(skills) {
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/account/applicant/script.js:96]: is this method repeated from the other js?

File: walk-in-interview/src/main/webapp/job-details/script.js [hunk 6]
```diff
@@ -198,6 +197,32 @@ function getJobId() {
   return urlParams.get(JOB_ID_PARAM);
 }
 
+const interestButton = document.getElementById('interest');
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/job-details/script.js]: Maybe these functionality can be in a different PR?

File: walk-in-interview/src/main/webapp/account/applicant/edit/index.html (hunk 1 above)
Comment (SibingWu) [reply walk-in-interview/src/main/webapp/account/applicant/edit/index.html:29]: the below skill list is just a hidden template which will be utilized in the `<ul id="skills"></ul>` above

File: walk-in-interview/src/main/webapp/account/applicant/script.js (hunk 5 above)
Comment (SibingWu) [reply walk-in-interview/src/main/webapp/account/applicant/script.js:96]: not really, since the `skills` in other js (job post related) is a dict, while here in account is an array

File: walk-in-interview/src/main/webapp/account/applicant/edit/script.js (hunk 2 above)
Comment (SibingWu) [reply walk-in-interview/src/main/webapp/account/applicant/edit/script.js:82]: we put the full applicant detail json in the url
